File: python/medium/Spiral_Matrix_II.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        answer = [[] for i in range(n)]
        num = 1
        for r in range(int(n/2+0.5)):
            x = r
            y = r
            for w in range(0, n-2*r):
                answer[y].insert(w+r, num)
                x+=1
                num+=1
            for l in range(1, n-2*r):
                y+=1
                answer[y].insert(-r, num)
                num+=1
            for w in range(1, n-2*r):
                x-=1
                answer[y].insert(r, num)
                num+=1
            for l in range(1, n-2*r-1):
                y-=1
                answer[y].insert(r, num)
                num+=1
        return answer

File: cpp/medium/Spiral_Matrix_II.cpp
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        vector<vector<int>> answer;
        for (int i = 0; i < n; i++) {
            vector<int> v = {};
            for (int j = 0; j < n; j++) {
                v.push_back(3*i+j);
            }
            answer.push_back(v);
        }

        int num = 1;
        int i = 0, r = 0;
        int count = n*n+1;
        while(true) {
            for (i = 0+r; i <= n-r-1; i++) {
                // cout << i << "," << r << "->" << num << endl;
                answer[r][i] = num;
                num++;
            }
            if (num == count) break;
            num--;
            for (i = 0+r; i <= n-r-1; i++) {
                // cout << n-1-r << "," << i << "->" << num << endl;
                answer[i][n-1-r] = num;
                num++;
            }
            if (num == count) break;
            num--;
            for (int i = n-r-1; i > r; i--) {
                // cout << i << "," << n-1-r << "->" << num << endl;
                answer[n-1-r][i] = num;
                num++;
            }
            if (num == count) break;
            for (int i = n-r-1; i >= 1+r; i--) {
                // cout << r << "," << i << "->" << num << endl;
                answer[i][r] = num;
                num++;
            }
            if (num == count) break;
            r++;
        }
        return answer;
    }
};
